### PR TITLE
feat: add token component

### LINKED ISSUE #4: Implement Add Token Button and Form Components
## Description
Develop a fully functional Add token and form components as per the design provided in the [FIGMA FILE](https://www.figma.com/design/pL9HcD5saTR6TtMsZYyS7z/STARKNET-UI-KITS?t=gT1lFVCNijoXRGlC-0) along with all it’s variants. The button  pops open a form for adding new tokens to a connected wallet.

## Requirements

- Design: Follow the provided Figma design precisely for both light and dark mode. Ensure that the component is fully responsive across different screen sizes.
- Style Guide: Adhere strictly to the project’s existing style guide for consistency in UI and UX.
- Functionality: The component should be fully functional, allowing users to seamlessly switch networks.

After building the component, create a sub-route in the `preview` route for the component and display each of it’s variant for easy testing. 

## Guidance
There is an already implemented Connect Button component which can serve as a reference for styling, functionality and previewing.

## Note
No other external dependencies should be installed other than the ones already provided.

### PR DESCRIPTION
**Pull request by Jemiiah**
resolves #4

### CODE REVIEW

File: app/components/add-token/AddToken.tsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+              onChange={(e) => {
+                const value = e.target.value;
+                if (/^\d*\.?\d*$/.test(value)) {
+                  setDecimals(value);
+                }
+              }}
+            />
+          </div>
+
+          <button
```
Comment (Oshioke-Salaki) [inline app/components/add-token/AddToken.tsx:115]: Disable when contract address is not available

File: app/preview/add-token/page.tsx:Zone.Identifier
Comment (stephanniegb) [inline app/preview/add-token/page.tsx:Zone.Identifier:1]: Please remove unnecessary files from the PR

**Comment by Oshioke-Salaki** [comment]
Can't review light mode, please look into it

**Comment by Jemiiah** [comment]
altight @Oshioke-Salaki and @stephanniegb

**Pull request merged**